app/src/main/java/com/example/evently/data/AccountDB.java

```diff
@@ -111,17 +113,59 @@ private Promise<Void> updateField(String email, String field, Object newValue) {
         return promise(docRef.update(field, newValue));
     }
 
+    /**
+     * The following method checks if an account is an admin account
+     * @param email The email of the account being verified
+     * @return A boolean about if the account is admin or not.
+     */
+    public Promise<Boolean> isAdmin(String email) {
+        return promise(db.collection("admin").document(email).get()).map(DocumentSnapshot::exists);
+    }
+
+    /**
+     * The following method sets an account as an Admin, only to be used for testing.
+     * For non-testing, admins should be created through the firebase console
+     * @param email The email of the account being turned admin
+     * @return A promise that the account will become an admin account.
+     */
+    @TestOnly
+    public Promise<Void> setAdmin(String email) {
+
+        /**
+         * The following 2 lines of code are based on a response from the LLM Claude Sonnet 4.5 by
+         * Anthropic: "how to store only document IDs in firebase from android with java? No data
+         * needs to be stored. Only the documentID"
+         *
+         * According to the response, we need to add at least one field because Firestore does not
+         * support empty documents.
+         */
+        HashMap<String, Object> emptyHashMap = new HashMap<>();
+        emptyHashMap.put("exists", true);
+
+        // Creates the document in the admin account list, saying that it exists.
+        return promise(db.collection("admin").document(email).set(emptyHashMap));
+    }
+
     /**
      * Nuke the accounts collection and all associated data.
      */
     @TestOnly
     public Promise<Void> nuke() {
-        return promise(accountsRef.get()).then(docs -> {
-            WriteBatch batch = FirebaseFirestore.getInstance().batch();
-            for (var doc : docs) {
-                batch.delete(doc.getReference());
-            }
-            return promise(batch.commit());
-        });
+        return promise(accountsRef.get())
+                .then(docs -> {
+                    WriteBatch batch = FirebaseFirestore.getInstance().batch();
+                    for (var doc : docs) {
+                        batch.delete(doc.getReference());
+                    }
+                    return promise(batch.commit());
+                })
+                // Also nukes the list of admins
+                .alongside(promise(db.collection("admin").get()).then(docs -> {
+                    WriteBatch batch = FirebaseFirestore.getInstance().batch();
+                    for (var doc : docs) {
+                        batch.delete(doc.getReference());
+                    }
+                    return promise(batch.commit());
+                }));
```

Comment:
```suggestion
        return promise(accountsRef.get())
                .then(docs -> {
                    WriteBatch batch = FirebaseFirestore.getInstance().batch();
                    for (var doc : docs) {
                        batch.delete(doc.getReference());
                    }
                    return promise(batch.commit());
                })
                // Also nukes the list of admins
                .alongside(promise(db.collection("admin").get()).then(docs -> {
                    WriteBatch batch = FirebaseFirestore.getInstance().batch();
                    for (var doc : docs) {
                        batch.delete(doc.getReference());
                    }
                    return promise(batch.commit());
                }));
```

This is likely better. Though for whatever reason, concatenating two lists in Java is horrid.

Reply:
Can't believe concatenating two iterables and iterating through them (i.e the stream thing I wrote) takes that much ugly code....

Reply:
It's a massive improvement over the code I wrote lol😂

```
        return promise(accountsRef.get())
                .then(docs -> {
                    WriteBatch batch = FirebaseFirestore.getInstance().batch();
                    for (var doc : docs) {
                        batch.delete(doc.getReference());
                    }
                    return promise(batch.commit());
                })
                // Also nukes the list of admins
                .alongside(promise(db.collection("admin").get()).then(docs -> {
                    WriteBatch batch = FirebaseFirestore.getInstance().batch();
                    for (var doc : docs) {
                        batch.delete(doc.getReference());
                    }
                    return promise(batch.commit());
                }));
```